src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -533,8 +533,7 @@ default Result submitResult(Result result, Exercise exercise) {
         double bonusPoints = Optional.ofNullable(exercise.getBonusPoints()).orElse(0.0);
 
         // Exam results and manual results of programming exercises and example submissions are always to rated
-        if (exercise.isExamExercise() || exercise instanceof ProgrammingExercise
-                || (result.getSubmission().isExampleSubmission() != null && result.getSubmission().isExampleSubmission())) {
+        if (exercise.isExamExercise() || exercise instanceof ProgrammingExercise || (result.isExampleResult() != null && result.isExampleResult())) {
```

Comment:
Is the check `result.isExampleResult() != null` really needed here? I think if you only use result.isExampleResult() should still be ok

Reply:
This would lead to an exception if `isExampleResult()` returns null. But you could compare it with Boolean.TRUE if you want:
```suggestion
        if (exercise.isExamExercise() || exercise instanceof ProgrammingExercise || Boolean.TRUE.equals(result.isExampleResult()) {
```
Imo both ways are ok here.

Reply:
I did like that first, then I encounter with NullExeption :D I checked the database, for the regular submissions example_result is not set (null).

Reply:
oh that's good idea